test/upgrade/preupgrade.go

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package upgrade
+
+import (
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+)
+
+// BrokerPreUpgradeTest tests channel operations before upgrade.
+func BrokerPreUpgradeTest() pkgupgrade.Operation {
+	return pkgupgrade.NewOperation("BrokerPreUpgradeTest", func(c pkgupgrade.Context) {
+		runBrokerSmokeTest(c.T)
+	})
+}
+
+// SourcePreUpgradeTest tests source operations before upgrade.
+func SourcePreUpgradeTest() pkgupgrade.Operation {
```

Comment:
I suppose the SourcePreUpgradeTest and SourcePostUpgradeTest are not required? We're testing the Kafka Broker and not Kafka Source here. Or am I missing something? The function looks empty anyway: https://github.com/knative-sandbox/eventing-kafka-broker/pull/617/files#diff-f82f3999b3c0ec8c23dd03ab628ef3a3deac24d945f10cbd9b637a3583268faaR41

Reply:
Another option is to "skip" this test. Make it clear that it's not implemented rather than adding TODO where the test would still be passing but not actually run anything.

Reply:
I removed it for now.

I'll add it later as needed.